scene/gui/tree.cpp

```diff
@@ -3359,8 +3359,10 @@ void Tree::_go_up() {
 			return;
 		}
 
-		select_single_item(prev, get_root(), selected_col);
-		queue_redraw();
+		if (selected_item) {
```

Comment:
Adding this condition makes the behavior different for multi-select and single-select mode. In single select mode, going up select the last node.
I think the real problem is that selected_col is -1 if nothing is selected. 
![image](https://github.com/godotengine/godot/assets/81109165/99d92a57-3db4-4da2-a86a-fb9f7e801346)

I suggest moving `int col = selected_col < 0 ? 0 : selected_col;` before the `if (select_mode == SELECT_MULTI) {` and using `col` as the third parameter of `select_single_item`.

Reply:
This is the approach taken by https://github.com/godotengine/godot/pull/92959 . I'm kinda confused why you want to make my PR exactly like theirs rather focus on their PR and see how it can be improved given it already has the approach you want.